include/CoreDisTools/coredistools.h

```diff
@@ -0,0 +1,36 @@
+//===--------- coredistools.h - Dissassembly tools for CoreClr
+//-------------===//
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// \brief Core Disassembly Tools API for CoreClr AOT/JIT
+///
+//===----------------------------------------------------------------------===//
+
+#if !defined(LLILC_TOOLS_COREDISTOOLS)
+#define LLILC_TOOLS_COREDISTOOLS
+
+#if !defined(DllIface)
+#define DllIface extern "C" __declspec(dllimport)
+#endif // defined(DllIface)
+
+enum TargetArch {
+  Target_Host,
```

Comment:
I believe there's a subtle ambiguity with `Target_Host` on Linux/ARM since there both ARM and Thumb code can exist in the same process.  This may not matter on Windows since generally only Thumb is allowed, though the concept is not unheard of even on Windows since I believe even Windows mixes x86 and x64 under WoW.

One option would be to have a major/minor architecture parameter, similar to how LLVM classifies things, but I think it's probably simpler to just make Thumb a peer of ARM for our purposes.

NOTE that I'd also probably be ok with just ignoring this issue under the assumption that this library is purpose-built for CoreCLR, which doesn't acknowledge the possibility of more than one simultaneous target architecture.  In that case, what you've done makes perfect sense, though I would probably rename Target_Arm to Target_Thumb to allow for unambiguous future expansion.